File: zsh/lss.lua
local lfs = require("lfs")

local folders = {}
local files = {}

local colors = {
    blue = "\27[34m",
    white = "\27[37m",
    reset = "\27[0m"
}

for item in lfs.dir(".") do
    if item ~= "." and item ~= ".." then
        local attr = lfs.attributes(item)
        if attr.mode == "directory" then
            table.insert(folders, item)
        else
            table.insert(files, item)
        end
    end
end

table.sort(folders)
table.sort(files, function(a, b)
    local extA = a:match("%.([^%.]+)$") or ""
    local extB = b:match("%.([^%.]+)$") or ""
    if extA == extB then
        return a < b
    end
    return extA < extB
end)

for _, f in ipairs(folders) do
    print(colors.blue .. f .. colors.reset)
end

for _, f in ipairs(files) do
    print(colors.white .. f .. colors.reset)
end

File: zsh/lss.py
from pathlib import Path
import sys

folders = []
files = []

colors = {
    "blue": "\033[34m",
    "white": "\033[37m",
    "reset": "\033[0m"
}

if __name__ == "__main__":
    path = Path(sys.argv[1])
    folders = []
    files = []

    for i in path.iterdir():
        if i.is_dir():
            folders.append(i.name)
        else:
            files.append(i.name)

    folders.sort()
    files.sort()

    for i in folders:
        print(f"{colors['blue']}{i}{colors['reset']}")

    for i in files:
        print(f"{colors['white']}{i}{colors['reset']}")
